modules/localization/msf/msf_localization.cc

```diff
@@ -294,6 +302,8 @@ void MSFLocalization::OnGnssHeading(
   localization_integ_.GnssHeadingProcess(*gnss_heading_msg);
 }
 
+void MSFLocalization::OnGps() { OnRawImu(raw_imu_msg_); }
```

Comment:
From my understanding, you are doing:
1. Cache latest IMU message.
2. OnGps() triggers processing the cached IMU message.
So that we are processing IMU at GPS's frequency.

I'm not sure if we should do the frequency control from sender side or receiver side. Let's wait for @Capri2014 's comment.

Reply:
> From my understanding, you are doing:
> 
> 1. Cache latest IMU message.
> 2. OnGps() triggers processing the cached IMU message.
>    So that we are processing IMU at GPS's frequency.
> 
> I'm not sure if we should do the frequency control from sender side or receiver side. Let's wait for @Capri2014 's comment.

Yes, change on receiver as of now, this is by no means ideal but has least side effect theoretically. (Still need to verify via offline dataset and on vehicle performance.)

Reply:
Is a lock necessary?

Reply:
> Is a lock necessary?

OK